express-error-handling/server/index.js

```diff
@@ -0,0 +1,169 @@
+require('dotenv/config');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error'); // eslint-disable-line
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      next(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```

Comment:
In the catch callback we only need to call `next(err)` that will pass control to the error middleware, so we don't need to also send a response here.

```JAVASCRIPT
.catch(err => {
  next(err);
});
```